lib/phoenix_live_reload/frame.ex

```diff
@@ -16,23 +16,22 @@ defmodule Phoenix.LiveReload.Frame do
     |> send_resp(200, """
       <html><body>
       <script>
-        #{phoenix_js()}
+        #{phoenix_js}
+
         var phx = require("phoenix")
         var socket = new phx.Socket("#{url}")
-        socket.connect()
-        socket.join("phoenix:live_reload", {})
-          .receive("ok", function(chan){
-            chan.on("assets_change", function(msg){
-              chan.off("assets_change")
-              window.top.location.reload()
-            })
-          })
+
+        #{phoenix_live_reload_js}
       </script>
       </body></html>
     """)
   end
 
-  defp phoenix_js() do
+  defp phoenix_js do
     File.read! Application.app_dir(:phoenix, "priv/static/phoenix.js")
   end
+
+  defp phoenix_live_reload_js do
+    File.read! Application.app_dir(:phoenix_live_reload, "priv/static/phoenix_live_reload.js")
```

Comment:
Ok sounds good!



> On Apr 24, 2015, at 4:12 PM, Chris McCord <notifications@github.com> wrote:
> 
> In lib/phoenix_live_reload/frame.ex:
> 
> >      File.read! Application.app_dir(:phoenix, "priv/static/phoenix.js")
> >    end
> > +
> > +  defp phoenix_live_reload_js do
> > +    File.read! Application.app_dir(:phoenix_live_reload, "priv/static/phoenix_live_reload.js")
> let's see if we can bake these into the module at compile time:
> 
> @phoenix_js File.read!(Application.app_dir(:phoenix, "priv/static/phoenix.js"))
> @phoenix_live_reload_js File.read!(Application.app_dir(:phoenix_live_reload, "priv/static/phoenix_live_reload.js"))
> Otherwise, we're all set!
> 
> —
> Reply to this email directly or view it on GitHub.
>